apps/core/src/components/Inputs/AddressInput.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { Input, InputType } from '@iota/apps-ui-kit';
+import { Close } from '@iota/ui-icons';
+import { useIotaAddressValidation } from '../../hooks';
+import React, { useCallback, useEffect } from 'react';
+import { useField, useFormikContext } from 'formik';
+
+export interface AddressInputProps {
+    name: string;
+    disabled?: boolean;
+    placeholder?: string;
+    label?: string;
+}
+
+export function AddressInput({
+    name,
+    disabled,
+    placeholder = '0x...',
+    label = 'Enter Recipient Address',
+}: AddressInputProps) {
+    const { validateField } = useFormikContext();
+    const [field, meta, helpers] = useField<string>(name);
+    const iotaAddressValidation = useIotaAddressValidation();
+
+    const formattedValue = iotaAddressValidation.cast(field.value);
+
+    const handleOnChange = useCallback(
+        (e: React.ChangeEvent<HTMLInputElement>) => {
+            const address = e.currentTarget.value;
+            helpers.setTouched(true);
+            helpers.setValue(iotaAddressValidation.cast(address));
```

Comment:
```suggestion
            helpers.setValue(iotaAddressValidation.cast(address)).then(() => {
                validateField(name);
            });
```
in this way we validate just this field in question doing something like this helpers.setValue(iotaAddressValidation.cast(address), true) would validate the entire form

Reply:
damn it, I didn't realize setValue returned a promise

Reply:
we should do something cleaner tho

```ts
await helpers.setValue(iotaAddressValidation.cast(address))
validateField(name);
```